### PR TITLE
UI: Fix the Add / Remove Vmware DC button on Zone detail view

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR fixes https://github.com/apache/cloudstack/issues/8773

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan ui

**Comment by Pearl1594** [comment]
Seems to show the button for non-vmware env - will fix it.

**Comment by Pearl1594** [comment]
@blueorangutan ui

File: ui/src/views/AutogenView.vue [hunk 1]
```diff
@@ -986,6 +986,12 @@ export default {
         }
         this.itemCount = apiItemCount
 
+        if (this.dataView && this.$route.path.includes('/zone/') && 'listVmwareDcs' in this.$store.getters.apis) {
```
Comment (rohityadavcloud) [inline ui/src/views/AutogenView.vue:989]: Does this need to be in autogenview or in zone specific detail view/tab ?

Comment (Pearl1594) [reply ui/src/views/AutogenView.vue:989]: we don't have an explicit component to control viewing the specific action. Hence, this updates the resource value before the zone.js  (https://github.com/apache/cloudstack/blob/main/ui/src/config/section/infra/zones.js#L237) file consumes it to determine whether or not to show the specific button.

**Review by sureshanaparti — APPROVED** [review]
LGTM

**Review by vishesh92 — APPROVED** [review]
clgtm. tested manually.
Maybe add a small comment explaining what we are doing here.

**Pull request merged**